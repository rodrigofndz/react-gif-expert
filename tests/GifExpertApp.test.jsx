import { fireEvent, render, screen } from "@testing-library/react";
import { GifExpertApp } from "../src/GifExpertApp";

describe('Tests in <GifExpertApp />', () => {
    test('Should render the app correctly', () => {
        const {container} = render(<GifExpertApp />);

        // Assert that the app renders without any errors
        expect(container).toMatchSnapshot();

        // Assert that the app initially has no gifs
        expect(screen.queryByRole('img')).not.toBe();
    });
    
    test('Should add a new category when the form is submitted', () => {
        render(<GifExpertApp />);
        
        // Get the input element and submit button
        const input = screen.getByRole('textbox');
        
        // Type a new category name in the input
        const newCategory = 'Naruto';
        fireEvent.change(input, { target: { value: newCategory } });
        
        // Submit the form
        fireEvent.submit(input);
        
        // Assert that the new category is added to the app
        expect(screen.getByText(newCategory)).toBeTruthy();
        
        // Assert that the input is cleared after submitting
        expect(input.value).toBe('');
    });
});