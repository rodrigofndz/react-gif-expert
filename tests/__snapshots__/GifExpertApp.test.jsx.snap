// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Tests in <GifExpertApp /> Should render the app correctly 1`] = `
<div>
  <h1>
    GifExpertApp
  </h1>
  <form
    aria-label="form"
  >
    <input
      placecholder="Buscar gifs"
      type="text"
      value=""
    />
  </form>
  <h3>
    One Piece
  </h3>
  <h2>
    Cargando...
  </h2>
  <div
    class="card-grid"
  />
</div>
`;
